examples/native_class_redscript/Main.cpp

```diff
@@ -0,0 +1,88 @@
+#include <RED4ext/RED4ext.hpp>
+#include <RED4ext/Scripting/IScriptable.hpp>
+#include <RED4ext/RTTITypes.hpp>
+
+struct MyCustomClass : RED4ext::IScriptable
+{
+    RED4ext::CClass* GetNativeType();
+};
+
+RED4ext::TTypedClass<MyCustomClass> cls("MyCustomClass");
+
+RED4ext::CClass* MyCustomClass::GetNativeType()
+{
+    return &cls;
+}
+
+void GetNumber(RED4ext::IScriptable* aContext, RED4ext::CStackFrame* aFrame, float* aOut, int64_t a4)
+{
+    MessageBox(nullptr, L"Test", L"Hello", MB_OK);
+    aFrame->code++; // skip ParamEnd
+    *aOut = 6.25;
+}
+
+RED4EXT_C_EXPORT void RED4EXT_CALL RegisterTypes()
+{
+    cls.flags = {.isNative = true};
+    RED4ext::CRTTISystem::Get()->RegisterType(&cls, 100000);
```

Comment:
Here might be nice to use `RED4ext::CRTTISystem::Get()->RegisterType(&cls)` so it gets an ID assigned automatically.